sky/serve/autoscalers.py

```diff
@@ -129,96 +134,99 @@ def collect_request_information(
-            scale_target = requests_per_replica / self.upper_threshold
-            target_num_replicas = int(scale_target * num_replicas)
-        elif (self.lower_threshold is not None and
-              requests_per_replica < self.lower_threshold):
-            scale_target = requests_per_replica / self.lower_threshold
-            target_num_replicas = int(scale_target * num_replicas)
-
+            self.request_timestamps) / self.qps_window_size
+        target_num_replicas = math.ceil(num_requests_per_second /
+                                        self.target_qps_per_replica)
         target_num_replicas = max(self.min_replicas,
                                   min(self.max_replicas, target_num_replicas))
+        logger.info(f'Requests per second: {num_requests_per_second}, '
+                    f'Current target number of replicas: {target_num_replicas}')
 
-        logger.info(f'Target number of replicas: {target_num_replicas}, '
-                    f'min_replicas: {self.min_replicas}, '
-                    f'max_replicas: {self.max_replicas}')
-
-        num_replicas_delta = 0
-        if num_replicas < self.min_replicas:
-            num_replicas_delta = self.min_replicas - num_replicas
-            self.upscale_counter = 0
-        elif num_replicas > self.max_replicas:
-            num_replicas_delta = self.max_replicas - num_replicas
-            self.downscale_counter = 0
-        elif target_num_replicas > num_replicas:
+        if not self.bootstrap_done:
+            self.bootstrap_done = True
+            return target_num_replicas
+        elif target_num_replicas > self.target_num_replicas:
             self.upscale_counter += 1
             self.downscale_counter = 0
             if self.upscale_counter >= self.scale_up_consecutive_periods:
                 self.upscale_counter = 0
-                num_replicas_delta = target_num_replicas - num_replicas
-        elif target_num_replicas < num_replicas:
+                return target_num_replicas
+        elif target_num_replicas < self.target_num_replicas:
             self.downscale_counter += 1
             self.upscale_counter = 0
             if self.downscale_counter >= self.scale_down_consecutive_periods:
                 self.downscale_counter = 0
-                num_replicas_delta = target_num_replicas - num_replicas
+                return target_num_replicas
         else:
             self.upscale_counter = self.downscale_counter = 0
-        logger.info(f'Upscale counter: {self.upscale_counter}/'
-                    f'{self.scale_up_consecutive_periods}. '
-                    f'Downscale counter: {self.downscale_counter}/'
-                    f'{self.scale_down_consecutive_periods}')
+        return self.target_num_replicas
 
-        if num_replicas_delta == 0:
+    def evaluate_scaling(
+        self,
+        replica_infos: List['replica_managers.ReplicaInfo'],
+    ) -> List[AutoscalerDecision]:
+
```

Comment:
It would be nice to add a docstr for this function to describe the behavior

Reply:
Sounds good.